### PR TITLE
Unify rar installation across distros and arches

### PR DESCRIPTION
**Pull request by flying-sausages**
## Fixes issues: 
- Ubuntu has `rar` in sources, Debian has to have a repo enabled, and `arm64` is a whole other mess
- Installs are killed if `rar` is not available

## Proposed Changes:
- Unify function for `rar` installation
- Explicitly handle each separate OS-Arch case
- Build `unrar` from source on un-handled architectures
- Explicitly mention when `rar` might not be available **but allow installations to continue**
- ~~Manage `rar` install status through lockfile~~
- ~~Add installer for `rar` (and `unrar`)~~
- ~~Add remover for `rar` (and `unrar`)~~

## Categories

- ~~New feature (non-breaking change which adds functionality)~~
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Some existing `functions` got changed (e.g. `utils`, `os`, `apt`, `ask`, etc.)

## Architectures

- `x86_64` is specifically handled as:
    - [x] Success
    - [ ] Error
    - Notes: Largely unchanged
- `arm64` is specifically handled as:
    - [x] Success
    - [ ] Error
    - Notes: `unrar` only (either via `apt` or installed from source)
- [x] Changes are arch agnostic 
    - Notes: `unrar` installed from source if no other combination is found

## Checklist

- [x] Docs have been made OR are not necessary
    - PR link: 
- [x] Changes to panel have been made OR are not necessary
    - PR link: 
- [x] Code is formatted [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Shellcheck isn't screaming [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Prints to terminal are handled [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#printing-into-the-terminal)
- [x] I have commented my code, particularly in hard-to-understand areas

## Testing done
OS & Version: Ubuntu for arm

### How have you tested this

`unrar` installs fine on `arm64` ubuntu
`unrar` builds from source on the same arch too

### CODE REVIEW

**Comment by liaralabs** [comment]
So I'm looking this over because in the current state, sabnzbd will fail to install rar.

That said, I don't necessarily agree with this being its own separate package entirely given that this package is dependency, but perhaps other folks have opinions on this. However, the package list is growing *insanely* long and this just further clutters it up with an application which should just be seamlessly installed if required by a package.

**Comment by flying-sausages** [comment]
Alright, I'll remove the installers and leave it as a function.

**Comment by flying-sausages** [comment]
> However, the package list is growing _insanely_ long and this just further clutters it up with an application which should just be seamlessly installed if required by a package.

Actually the way this is designed, the same function will just quietly install if the dependency is missing. The same function that installers called before is now handling more cases. Nothing will fail if it requires `rar` and the `rar` package isn't installed beforehand.

It just opens the paths to remove the package, and install it without having to install the depndee for it. This would allow OSs without `rar`/`unrar` packaged to install it.

**Comment by flying-sausages** [comment]
I take the last comments back lol let's just not change the way it's managed rn

**Comment by flying-sausages** [comment]
Success on ubuntu for arm when installing rutorrent, should be tested on debian x64 still and then it's good to merge
afbeelding

File: scripts/update/10-dependencies.sh [hunk 1]
```diff
@@ -17,3 +17,19 @@ if [[ ${missing[0]} != "" ]]; then
...
+    fi
+    if ! grep -q non-free /etc/apt/sources.list; then
+        echo_info "Enabling non-free repo"
+        apt-add-repository non-free >> ${log} 2>&1
+        trigger_apt_update=true
+    fi
+    if [[ $trigger_apt_update == "true" ]]; then
+        apt_update
+    fi
+fi
```
Comment (flying-sausages) [inline scripts/update/10-dependencies.sh:35]: Should this be above the actual dependency installation?

Comment (liaralabs) [reply scripts/update/10-dependencies.sh:35]: `apt-add-repository` will not be present on most Debian installs, thus it needs to be done after the deps are checked. I don't think any of the global deps themselves should be coming from non-free or contrib

Comment (flying-sausages) [reply scripts/update/10-dependencies.sh:35]: 👍 thanks

**Pull request merged**